src/index.ts

```diff
@@ -2363,6 +2373,152 @@ const processHasUncaughtExceptionCaptureCallback = staticCall(
 const processEmitWarning = bind.bind(ogProcess.emitWarning, ogProcess)
 const processDebugPort = Number(ogProcess.debugPort)
 
+const createSafeIterator = <T, TReturn, TNext>(
+  factory: (self: T) => IterableIterator<T>,
+  next: (...args: [] | [TNext]) => IteratorResult<T, TReturn>
+) => {
+  class SafeIterator implements IterableIterator<T> {
+    _iterator: any
+    constructor(iterable: T) {
+      this._iterator = factory(iterable)
+    }
+    next() {
+      return next(this._iterator)
+    }
+    [Symbol.iterator]() {
+      return this
+    }
+    [SymbolIterator]() {
+      return this
+    }
+  }
+  ObjectSetPrototypeOf(SafeIterator.prototype, null)
+  ObjectFreeze(SafeIterator.prototype)
+  ObjectFreeze(SafeIterator)
+  return SafeIterator
+}
+
+const copyProps = (src: object, dest: object) => {
+  ArrayPrototypeForEach(ReflectOwnKeys(src), key => {
+    if (!ReflectGetOwnPropertyDescriptor(dest, key)) {
+      ReflectDefineProperty(dest, key, {
+        __proto__: null,
+        ...ReflectGetOwnPropertyDescriptor(src, key),
+      } as PropertyDescriptor)
+    }
+  })
+}
+
+interface Constractable<T> extends NewableFunction {
+  new (...args: any[]): T
+}
+const makeSafe = <T, C extends Constractable<any>>(unsafe: C, safe: C) => {
+  if (SymbolIterator in unsafe.prototype) {
+    const dummy = new unsafe()
+    let next // We can reuse the same `next` method.
+
+    ArrayPrototypeForEach(ReflectOwnKeys(unsafe.prototype), key => {
+      if (!ReflectGetOwnPropertyDescriptor(safe.prototype, key)) {
+        const desc = ReflectGetOwnPropertyDescriptor(unsafe.prototype, key)
+        if (
+          desc &&
+          typeof desc.value === 'function' &&
+          desc.value.length === 0 &&
+          SymbolIterator in
+            (FunctionPrototypeCall(desc.value, dummy) ?? {})
+        ) {
+          const x: string[] = []
+          const y = x[Symbol.iterator]
+
+          const createIterator = uncurryThis(desc.value) as unknown as (
+            val: T
+          ) => IterableIterator<any>
+          next ??= uncurryThis(createIterator(dummy).next)
```

Comment:
Not supported on node 14, so this is a breaking change. It's a function, so `||=` is equivalent.
```suggestion
          next ||= uncurryThis(createIterator(dummy).next)
```